scopes/component/new-component-helper/new-component-helper.main.runtime.ts

```diff
@@ -31,16 +32,44 @@ export class NewComponentHelperMain {
     return ComponentID.fromObject({ name: fullComponentName }, scope);
   }
 
+  /**
+   * Synchronously checks if the given directory contains any files.
+   */
+  private dirContainsFiles(dirPath: string): boolean {
+    if (!fs.existsSync(dirPath)) return false;
+
+    const contents = fs.readdirSync(dirPath);
+    for (const item of contents) {
+      if (fs.statSync(path.join(dirPath, item)).isFile()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   /**
    * when creating/forking a component, the user may or may not provide a path.
    * if not provided, generate the path based on the component-id.
    * the component will be written to that path.
    */
-  getNewComponentPath(componentId: ComponentID, pathFromUser?: string): PathLinuxRelative {
-    if (pathFromUser) return pathFromUser;
+  getNewComponentPath(componentId: ComponentID, pathFromUser?: string, componentsToCreate?: number): PathLinuxRelative {
+    if (pathFromUser) {
+      const fullPath = path.join(this.workspace.path, pathFromUser);
+      const componentPath = componentId.namespace?.length
+        ? path.join(componentId.namespace, componentId.name)
```

Comment:
you probably can just use `componentId.fullName` no?

Reply:
yes I forgot the full name is already in the structure of a dir :)